### ISSUE TITLE
Cannot specify GPU to train the model, only GPU 0 can be used

### ISSUE THREAD

**Issue by lfbcccc** [open-mmlab/mmsegmentation#2960]
**Describe the bug**
Cannot specify GPU to train the model, only GPU 0 can be used

**Reproduction**

1. What command or script did you run?

train.py

3. Did you make any modifications on the code or config? Did you understand what you have modified?

Only when I change the Gpu_id to 1 will an error occur
![微信截图_20230426223716](https://user-images.githubusercontent.com/111995105/234611066-6ee74784-e33d-401b-88c8-9855de91c3e2.png)

RuntimeError: Expected tensor for 'out' to have the same device as tensor for argument #3 'batch2'; but device 1 does not equal 0 (while checking arguments for baddbmm)

**Comment by xiexinch** [comment]
Please refer to https://mmsegmentation.readthedocs.io/en/latest/user_guides/4_train_test.html#launch-multiple-jobs-on-a-single-machine

**Comment by lfbcccc** [comment]
When I use the following command

CUDA_VISIBLE_DEVICES=1 sh tools/dist_train.sh local_configs/segnext/tiny/segnext.tiny.512x512.ade.160k.py 1 --work-dir output/voc_coal_dilation_gam_big_1024_80k_3_gai 

This error may occur. 
train.py: error: unrecognized arguments: local_configs/segnext/tiny/segnext.tiny.512x512.ade.160k.py
Killing subprocess 172132
How can I resolve it
please

**Comment by xiexinch** [comment]
Did you make any modifications to the `train.py` script?

**Comment by lfbcccc** [comment]
Yes, modifications were made during single card training, mainly including
![1682583134430](https://user-images.githubusercontent.com/111995105/234802740-43eb5876-46e7-4739-b874-87a49c6f9ef6.jpg)

In addition, I would like to inquire if there is a method for directly specifying GPU training in train.py

**Comment by wujiang0156** [comment]
> Please refer to https://mmsegmentation.readthedocs.io/en/latest/user_guides/4_train_test.html#launch-multiple-jobs-on-a-single-machine

How to achieve single-machine multi-Gpus training on windows？
train models on win10 with multi-Gpus??
